src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -152,33 +166,56 @@ protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string n
             string identifier,
             Document document,
             SemanticModel semanticModel,
-            Func<SyntaxToken, SyntaxNode> findParentNode,
+            Func<SyntaxToken, SyntaxNode>? findParentNode,
             CancellationToken cancellationToken)
         {
             var symbolsMatch = GetStandardSymbolsMatchFunction(symbol, findParentNode, document.Project.Solution, cancellationToken);
+            return FindReferencesInDocumentUsingIdentifierAsync(
+                symbol, identifier, document, semanticModel, symbolsMatch, cancellationToken);
+        }
 
+        protected static Task<ImmutableArray<FinderLocation>> FindReferencesInDocumentUsingIdentifierAsync(
+            ISymbol symbol,
+            string identifier,
+            Document document,
+            SemanticModel semanticModel,
+            Func<SyntaxToken, SemanticModel, (bool matched, CandidateReason reason)> symbolsMatch,
+            CancellationToken cancellationToken)
+        {
+            var findInGlobalSuppressions = ShouldFindReferencesInGlobalSuppressions(symbol, out var getDocumentationCommentId);
```

Comment:
do we erally need the delegate here?  if we're finidng in global suppressions, why not just get the actual docid and pass that along?

Reply:
Done.